pgml-extension/src/config.rs

```diff
@@ -1,9 +1,36 @@
+use std::collections::BTreeMap;
+use std::env;
 use std::ffi::CStr;
+use std::path::PathBuf;
 
+use lazy_static::lazy_static;
+use pgrx::guc::GucFlags;
 #[cfg(any(test, feature = "pg_test"))]
 use pgrx::{pg_schema, pg_test};
+use pgrx::{GucContext, GucRegistry, GucSetting};
 use pgrx_pg_sys::AsPgCStr;
 
+// GUC variables for huggingface
+pub static CONFIG_CACHE: &str = "pgml.cache";
+pub static CONFIG_OFFLINE: &str = "pgml.offline";
+
+// environment variables for huggingface and sentence-transformers
+static ENV_HF_HOME: &str = "HF_HOME";
+static ENV_SENTENCE_TRANSFORMERS_HOME: &str = "SENTENCE_TRANSFORMERS_HOME";
+static ENV_HF_HUB_OFFLINE: &str = "HF_HUB_OFFLINE";
+
+struct Guc {
+    pgml_cache: GucSetting<Option<&'static CStr>>,
+    pgml_offline: GucSetting<bool>,
+}
+
+lazy_static! {
+    static ref GUC: Guc = Guc {
```

Comment:
Since this is now only used once at server start, this can be a transient variable to set the env, right? We don't need to keep it around in a lazy_static?

Reply:
The reason that we use functions in GucRegistry is that we can define bool GUC and add descriptions for GUC variables.